python/tank/authentication/user.py

```diff
@@ -118,6 +126,93 @@ def impl(self):
         return self._impl
 
 
+class ShotgunSamlUser(ShotgunUser):
+    """
+    This specialized shotgun user is needed when SSO is used, as it provides
+    mechanisms
+    """
+
+    def __init__(self, impl):
+        """
+        :param impl: Internal user implementation class this class proxies.
+        """
+        super(ShotgunSamlUser, self).__init__(impl)
+        self._timer = None
+
+    def get_claims_expiration(self):
+        """
+        Obtain claims expiration for the user.
+
+        :returns: The expiration in seconds since January 1st 1970 UTC.
+        """
+        # The import must be located here. If it is at global scope, there
+        # is an import error.
+        from .shotgun_shared import saml2_sso
```

Comment:
I just fixed this circular dependency in one of my feature branches. It will go in master in the coming week and we can revisit your imports at that point.